mmv1/products/dialogflowcx/TestCase.yaml

```diff
@@ -0,0 +1,451 @@
+      Format: projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>/testCases/<TestCase ID>.
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+  - !ruby/object:Api::Type::Array
+    name: 'tags'
+    description: |
+      Tags are short descriptions that users may apply to test cases for organizational and filtering purposes.
+      Each tag should start with "#" and has a limit of 30 characters
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    required: true
+    description: |
+      The human-readable name of the test case, unique within the agent. Limit of 200 characters.
+    validation: !ruby/object:Provider::Terraform::Validation
+      function: 'validation.StringLenBetween(0, 200)'
+  - !ruby/object:Api::Type::String
+    name: 'notes'
+    description: |
+      Additional freeform notes about the test case. Limit of 400 characters.
+    validation: !ruby/object:Provider::Terraform::Validation
+      function: 'validation.StringLenBetween(0, 400)'
+  - !ruby/object:Api::Type::NestedObject
+    name: 'testConfig'
+    description: |
+      Config for the test case.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: 'trackingParameters'
+        description: |
+          Session parameters to be compared when calculating differences.
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::String
+        name: 'flow'
+        description: |
+          Flow name to start the test case with.
+          Format: projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>/flows/<Flow ID>.
+          Only one of flow and page should be set to indicate the starting point of the test case. If neither is set, the test case will start with start page on the default start flow.
+        conflicts:
+          - test_config.0.page
+      - !ruby/object:Api::Type::String
+        name: 'page'
+        description: |
+          The page to start the test case with.
+          Format: projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>/flows/<Flow ID>/pages/<Page ID>.
+          Only one of flow and page should be set to indicate the starting point of the test case. If neither is set, the test case will start with start page on the default start flow.
+        conflicts:
+          - test_config.0.flow
+  - !ruby/object:Api::Type::Array
+    name: 'testCaseConversationTurns'
+    description: |
+      The conversation turns uttered when the test case was created, in chronological order. These include the canonical set of agent utterances that should occur when the agent is working properly.
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::NestedObject
+          name: 'userInput'
+          description: |
+            The user input.
+          properties:
+            - !ruby/object:Api::Type::NestedObject
+              name: 'input'
```

Comment:
It looks like `intent` and `audio` are missing here. This is already a pretty big PR so I'm fine with leaving them for later assuming that's intentional.

Reply:
Unfortunately, TestCases don't support `intent` or `audio`, per [the documentation](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/ConversationTurn#userinput). I've added a comment to this effect.